packages/fuels-programs/src/contract.rs

```diff
@@ -626,12 +613,11 @@ where
         let tx = self.build_tx().await?;
         let provider = self.account.try_provider()?;
 
-        let chain_id = provider.chain_id();
-        self.cached_tx_id = Some(tx.id(chain_id));
-
         let tx_status = if simulate {
             provider.checked_dry_run(tx).await?
         } else {
+            self.cached_tx_id = Some(tx.id(provider.chain_id()));
```

Comment:
Uh, haven't thought about it that way; that's a good point!

Reply:
You do not need simulate to have the same effect. You could do:
```rust
let mut handler = contract.methods().some_fn();
let response = handler.get_response(some_receipts);
```
`cached_tx_id` is `None` and that gets propagated to the response.

 The `some_receipts` here is crucial. How would you get the receipts ? Either you manually send a transaction or you `call` the the method and get the right receipts. My thought was: simulate should not be used to get a `tx_id`. The user would get a `FuelResponse` with the `tx_id` only if the `tx` was committed. If the user needs the `tx_id` there are other ways to get it.
 
I am open for suggestions :)

Reply:
I'd ideally keep the difference between running and dry running minimal. Refusing to cache the tx id while simulating creates an unnecessary change to the behavior. So if the user wants to introduce a `dry-run` switch to their logic they would need to be careful to get the tx id differently if they are simulating. If not, they'll get the id in the response.

> If the user needs the tx_id there are other ways to get it.

This can also be said for using `call`. The user could have gotten the id without extracting it from the response.

Reply:
Good point. I will revert that part.